src/server/verification/verificationAPI.js

```diff
@@ -212,7 +214,16 @@ const setup = (app: Router, verifier: VerificationAPI, storage: StorageAPI) => {
 
         if (verified === false) return
 
-        await storage.updateUser({ identifier: user.loggedInAs, smsValidated: true, mobile: tempSavedMobile })
+        await storage.updateUser({
+          identifier: user.loggedInAs,
+          smsValidated: true,
+          mobile: hashedNewMobile,
+          'otp.mobile': undefined
+        })
+        if (currentMobile && currentMobile !== hashedNewMobile) {
```

Comment:
could the indexing be implemented via Moongoose's middleware (pre/post save callbacks) ?
or consolidated in the user storage / separate service as you did with the claimQueue ?

Reply:
indexing happens on gundb not mongoose.
indexes are used on client side to find the profile of a blockchain address or to find the address of a mobile number (to send directly by phone)
first version indexes use global gundb scope which is writable by all, so someone can change the index maliciously.
in this version there's a trusted index under the GoodDollar user scope, so only we have access to it.

Reply:
Okay. And, all user management logic on server is inside storageAPI because we don't have UserStorage like on DAPP (just  userDBPrivate which interacts with mongo only not with the Gun). So storageAPI is a better place to perform those actions (such as remove from old ondex / add to new).
Understood.